### DISCUSSION TITLE
Users in Custom Script

### DISCUSSION THREAD

**Discussion by therealpoys** [netbox-community/netbox#9539] [Q&A] [answered]
Hello,

I want to develop a Custom Script,where I need the Users in an "ObjectVar". 
Is this possible?
I tried it with : 
`
from extras.scripts import *
from django.contrib.auth.models import Group, User
# section name
name = 'Allocate contacts to devices'

class allocatecontacts2device(Script):

    class Meta:
        """
        defines script attributes
        """
        name = "Allocate contacts to devices"
        description = '''Script creates or updates a connection between 2 virtual machines in netbox.'''
        commit_default = False
        
    user = ObjectVar(
        label='user',
        description="user",
        model=User,
    )
`
Greetings,
Jannik

**Comment by bistraque** [comment, marked as answer]
My understanding of custom scripts is that ObjectVar only applies to netbox models. User being a django model, I doubt this is possible.

Looking at your example however, it seems that you are trying to use "User" in place of "Contact" which should be a legit netbox model in an ObjectVar script variable. Beware that using contacts implies using contact roles and contact assignments.

Syncing User table and Contact table is a totally different issue of course :)

**Reply by therealpoys** [reply]
Thank you!
It Works.

**Reply by jeremystretch** [reply]
FYI I'm pretty sure you should still be able to reference the User and/or Group models provided by Django, since there's a REST API endpoint for it. If it's not working, you may have run into a bug.
